bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -194,6 +194,24 @@ impl<'a> CircuitInputStateRef<'a> {
         Ok(())
     }
 
+    /// Push a read [`Operation`](crate::operation::Operation) to be into the
+    /// [`OperationContainer`](crate::operation::OperationContainer) with the
+    /// next [`RWCounter`](crate::operation::RWCounter) and then adds a
+    /// reference to the stored operation
+    /// ([`OperationRef`]) inside the
+    /// bus-mapping instance of the current [`ExecStep`]. Then increase the
+    /// block_ctx [`RWCounter`](crate::operation::RWCounter) by one.
+    pub fn read_op_reversible<T: Op>(&mut self, step: &mut ExecStep, op: T) -> Result<(), Error> {
+        let op_ref = self.block.container.insert(Operation::new_reversible(
+            self.block_ctx.rwc.inc_pre(),
+            RW::READ,
+            op,
+        ));
+        step.bus_mapping_instance.push(op_ref);
+
+        Ok(())
+    }
+
```

Comment:
I think this doesn't make too much sense, read doesn't mutate state so will never need reversion.

Reply:
just use `state.push_op(` is indeed more cheaper , will remove this new method.

Reply:
removed in [58beba4](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/827/commits/58beba4d5c114a534f898e68e1ec2434c44f7453) thanks !!